USF-2170 Content localization (#333)
commerce-docs/microsite-commerce-storefront: src/content/docs/merchants/multistore/content-localization.mdx

---
title: Content localization
description: Step-by-step guide for localizing the Edge Delivery content blocks of a storefront.
---

import { Tabs, TabItem, Steps } from '@astrojs/starlight/components';
import Tasks from '@components/Tasks.astro';
import Task from '@components/Task.astro';
import LinkCard from '@components/LinkCard.astro';
import Screenshot from '@components/Screenshot.astro';
import Callouts from '@components/Callouts.astro';
import OptionsTable from '@components/OptionsTable.astro';
import { Image } from 'astro:assets';
import Diagram from '@components/Diagram.astro';
import CardGrid from '@components/CardGrid.astro';
import Card from '@components/Card.astro';
import Aside from '@components/Aside.astro';
import { Icon } from '@astrojs/starlight/components';
import Link from '@components/Link.astro';
import Vocabulary from '@components/Vocabulary.astro';
import Embed from '@components/Embed.astro';

> This guide will walk you through the process of localizing your Edge Delivery content blocks of your storefront.

Reviewer: ```suggestion
This guide will walk you through the process of localizing the content of your storefront using DA.live.
```

This feature is about localizing content authored by DA.live. It's also not only about blocks; it's really about the document. For example, one could translate a whole document/page with no blocks. JSONs are also translated, so it's really about "content."

> ## Prerequisites

Reviewer: It’s not strictly a prerequisite for content translation, but to achieve full localization—including storefront setup (currency, translations, localized storefront links) and enabling the store switcher—developers need to follow the steps outlined at http://localhost:4321/merchants/multistore/.

- You have a storefront project in the Document Authoring app (<Link href="https://da.live" text="da.live" />) of the Storefront Builder.
- You know the basics of using the Document Authoring tool as described in the documentation here: (<Link href="https://da.live/docs" text="da.live/docs" />).

## Additional resources

- <Link href="https://www.aem.live/docs/translation-and-localization" text="Translation and Localization" />
- <Link href="https://da.live/docs/administration/translation" text="Translation, localization, and rollout" />

## Big Picture

The following diagram provides the basic workflow for translating the Edge Delivery content blocks of your storefront into multiple languages using the Document Authoring apps <Link href="https://da.live" text="da.live" />.

> 

Reviewer: ```suggestion
The following diagram provides the basic workflow for translating the Edge Delivery content of your storefront into multiple languages using the Document Authoring apps <Link href="https://da.live" text="da.live" />.

```

<Tasks>

<Task>
### Navigate to your storefront's Apps page

<Steps>
1. Open your storefront project in the Document Authoring tool (<Link href="https://da.live" text="da.live" />).
1. Select Apps from the left sidebar.

</Steps>

</Task>

<Task>
### Use the Traverse app to get a list of site URLs

<Steps>
1. Open the **Traverse** app by selecting its *Go* button.
1. On the Crawl Tree page, enter your storefront's organization name and site name. Example: `/acme/storefront`. Ensure that you enter a forward slash (`/`) at the beginning of the string or it will not work. 
1. Select **Crawl** button to start the crawl.
1. Select the **Copy list** button after the site's URLs have been traversed. This puts all the URLs in your clipboard.
1. Return to the Apps page.

</Steps>

</Task>

<Task>

### Use the Translation app to translate your storefront

<Steps>
1. Open the **Translate** app by selecting its *Go* button.
1. On the Localization page, enter the Title of the translation project: `acme-storefront-fr`.

> 1. Paste the list of URLs from the **Traverse** app into the **URLs** field.

Reviewer: Although it might not technically cause issues, merchants may prefer not to translate every file. We should recommend excluding certain files, such as `metadata.json`, `redirects.json`, and `sitemap.json`.

---
title: Content localization
description: Step-by-step guide for localizing the Edge Delivery content blocks of a storefront.
---

import { Tabs, TabItem, Steps } from '@astrojs/starlight/components';
import Tasks from '@components/Tasks.astro';
import Task from '@components/Task.astro';
import LinkCard from '@components/LinkCard.astro';
import Screenshot from '@components/Screenshot.astro';
import Callouts from '@components/Callouts.astro';
import OptionsTable from '@components/OptionsTable.astro';
import { Image } from 'astro:assets';
import Diagram from '@components/Diagram.astro';
import CardGrid from '@components/CardGrid.astro';
import Card from '@components/Card.astro';
import Aside from '@components/Aside.astro';
import { Icon } from '@astrojs/starlight/components';
import Link from '@components/Link.astro';
import Vocabulary from '@components/Vocabulary.astro';
import Embed from '@components/Embed.astro';

> This guide will walk you through the process of localizing the content of your storefront using DA.live.

Reviewer 2: ```suggestion
This guide will walk you through the process of localizing the content of your storefront using Document Authoring (da.live).
```

Reviewer 2: Your index file calls the product Document Author. Fix the discrepancy.

## Prerequisites

- You have a storefront project in the Document Authoring app (<Link href="https://da.live" text="da.live" />) of the Storefront Builder.
- You know the basics of using the Document Authoring tool as described in the documentation here: (<Link href="https://da.live/docs" text="da.live/docs" />).

:::note

> To achieve full localization—including storefront setup (currency, translations, localized storefront links) and enabling the store switcher—developers the developers for your storefront site need to follow the steps outlined in [multistore setup](/merchants/multistore/).

Reviewer 2: ```suggestion
To achieve full localization—including storefront setup (currency, translations, localized storefront links) and enabling the store switcher—the developers for your storefront site need to follow the steps outlined in [multistore setup](/merchants/multistore/).
```

<Diagram caption="Workflow for localizing your storefront">
  ![Workflow for localizing your storefront](../images/localization-steps.png)
</Diagram>

## Step-by-Step

These steps provide the details for the translation workflow shown in the diagram. The steps describe how to translate the default English store to another language. In this example, we will translate the default English store to French (`fr`).

<Tasks>

<Task>
### Navigate to your storefront's Apps page

<Steps>
1. Open your storefront project in the Document Authoring tool (<Link href="https://da.live" text="da.live" />).
1. Select Apps from the left sidebar.

</Steps>

</Task>

<Task>
### Use the Traverse app to get a list of site URLs

<Steps>
1. Open the **Traverse** app by selecting its *Go* button.
1. On the Crawl Tree page, enter your storefront's organization name and site name. Example: `/acme/storefront`. Ensure that you enter a forward slash (`/`) at the beginning of the string or it will not work. 

> 1. Select **Crawl** button to start the crawl.

Reviewer 2: ```suggestion
1. Select the **Crawl** button to start the crawl.
```

1. Select the **Copy list** button after the site's URLs have been traversed. This puts all the URLs in your clipboard.
1. Return to the Apps page.

</Steps>

</Task>

<Task>

### Use the Translation app to translate your storefront

<Steps>
1. Open the **Translate** app by selecting its *Go* button.
1. On the Localization page, enter the Title of the translation project: `acme-storefront-fr`.
1. Paste the list of URLs from the **Traverse** app into the **URLs** field.
   :::note

>    We recommend excluding certain files—such as metadata.json, redirects.json, and sitemap.json—from the translation process. Adding these files to the translation process shouldn't cause any technical issues with translation, but translating them is not necessary.

Reviewer 2: Should these files be in monospace?

   :::
1. Select the **Next** button.
1. View the validated URLs to ensure that they are all correct.
1. Select the **Next** button.
1. Select one or more languages you want to translate to. In this example, we will translate to French (`fr`).
1. For the French language, select **Translate** from the selector, then select the **Create project** button.
1. From the project page, select the **Send all for translation** button.

</Steps>

<Aside type="note">

> By default, the Translation app will use the Google Translate service to translate the content of the storefront into the selected language. Your organization can also connect other translation services for a more robust translation experience.

Reviewer 2: A link to the doc that explains how to configure translation services would be ideal.
